### PR TITLE
Add the `Span` of the `move` keyword to the HIR.

### PR DESCRIPTION
**Pull request by dnbln**
This is required to implement a lint like the one described here: https://github.com/rust-lang/rust-clippy/issues/11721

### CODE REVIEW

File: compiler/rustc_ast/src/ast.rs [hunk 1]
```diff
@@ -1548,7 +1548,10 @@ pub struct QSelf {
 #[derive(Clone, Copy, PartialEq, Encodable, Decodable, Debug, HashStable_Generic)]
 pub enum CaptureBy {
     /// `move |x| y + x`.
-    Value,
+    Value {
+        /// The span of the `move` keyword.
+        move_kw: Span,
```
Comment (cjgillot) [inline compiler/rustc_ast/src/ast.rs:1553]: The AST visitors need to visit this span.

Comment (dnbln) [reply compiler/rustc_ast/src/ast.rs:1553]: I'm not entirely sure I understand what that means, but I'm gonna guess that adding this to the `rustc_ast::visit::Visitor` would be enough? Correct me if I'm wrong please, this is the first time I touch rustc code (directly)😅 
```rust
    fn visit_capture_by(&mut self, capture_by: &'ast CaptureBy) {
        // Nothing to do
    }
```

Comment (cjgillot) [reply compiler/rustc_ast/src/ast.rs:1553]: The default implementation of `visit_capture_by` in `MutVisitor` should call `visit_span` on `move_kw`.

**Comment by dnbln** [comment]
Well it seems like the pipeline passed so I'm gonna mark this as ready for review now, but please mention me before merging so I can squash all the commits.

**Pull request merged**